manifests/init.pp

```diff
@@ -156,23 +156,34 @@
   $shared_with_foreman_vhost = !$pulpcore_mirror
 
   $rhsm_path = '/rhsm'
-  $rhsm_port = $reverse_proxy_real ? {
-    true  => $reverse_proxy_port,
-    false => 443
-  }
+  $rhsm_port = 443
 
   ensure_packages('katello-debug')
 
   include certs::foreman_proxy
   Class['certs::foreman_proxy'] ~> Service['foreman-proxy']
 
   if $reverse_proxy_real {
-    class { 'foreman_proxy_content::reverse_proxy':
+    foreman_proxy_content::reverse_proxy { 'katello-reverse-proxy':
       url  => "${foreman_url}/",
       port => $reverse_proxy_port,
     }
   }
 
+  if $pulpcore_mirror {
+    if $rhsm_port != $reverse_proxy_port {
+      $pulpcore_https_vhost_name = "rhsm-pulpcore-reverse-proxy-${rhsm_port}"
+
+      foreman_proxy_content::reverse_proxy { $pulpcore_https_vhost_name:
```

Comment:
I still would like to get rid of the generic "proxy / to Foreman" reverse proxy. I think there are dragons hiding there.